hummingbot/client/config/strategy_config_data_types.py

```diff
@@ -0,0 +1,171 @@
+from typing import Dict
+
+from pydantic import Field, validator
+
+from hummingbot.client import settings
+from hummingbot.client.config.config_data_types import BaseClientModel, ClientConfigEnum, ClientFieldData
+from hummingbot.client.config.config_validators import (
+    validate_exchange,
+    validate_market_trading_pair,
+    validate_strategy,
+)
+from hummingbot.client.settings import AllConnectorSettings, ConnectorType
+
+
+class BaseStrategyConfigMap(BaseClientModel):
+    strategy: str = Field(
+        default=...,
+        client_data=ClientFieldData(
+            prompt=lambda mi: "What is your market making strategy?",
+            prompt_on_new=True,
+        ),
+    )
+
+    @validator("strategy", pre=True)
+    def validate_strategy(cls, v: str):
+        ret = validate_strategy(v)
+        if ret is not None:
+            raise ValueError(ret)
+        return v
+
+
+class BaseTradingStrategyConfigMap(BaseStrategyConfigMap):
+    exchange: ClientConfigEnum(  # rebuild the exchanges enum
+        value="Exchanges",  # noqa: F821
+        names={e: e for e in
+               sorted([connector_setting.name for connector_setting
+                       in AllConnectorSettings.get_connector_settings().values()
+                       if connector_setting.type is ConnectorType.Exchange] + settings.PAPER_TRADE_EXCHANGES)},
```

Comment:
Why not use `AllConnectorSettings. get_exchange_names` instead?

Reply:
Good point. I made the change.